builtin/common/information_formspecs.lua

```diff
@@ -110,6 +119,78 @@ local function build_privs_formspec(name)
 end
 
 
+-- If 'modname' contains inside some modpack, returns name of that modpack, otherwise nil.
+local function get_modpack_name(modname)
+	local path = core.get_modpath(modname)
+	local dirs = path:split(DIR_DELIM)
+
+	if dirs[#dirs-2] == "mods" then
+		return dirs[#dirs-1]
```

Comment:
`modpack.conf` should be checked for the modpack name.

Reply:
It is impossible since the engine doesn't allow to read mods files from 'common', only from 'mainmenu'. The access is just blocked.

Reply:
Couldn't you read `modpack.conf` in as a `Settings` object?

Reply:
possibly relevant conversation:

https://irc.minetest.net/minetest/2022-10-13#i_6024802

the gist of it (i think) is that Settings is (currently) hard-coded to abide by the rules of the "secure" environment. i haven't tested, but i think that applies to "builtin" outside of the main menu. see #12857.

Reply:
I think reading from the file would work if it were done at load time: https://github.com/minetest/minetest/blob/02c293ec63c55b92be3257aac08308ee4fffbe1f/src/script/cpp_api/s_security.cpp#L561-L565

Reply:
Maybe permit for common to access the insecure environment on default as mainmenu? Where can I find the code that does that?

Reply:
That won't help until `Settings` is changed, but I think you can permit it like so:

```patch
diff --git a/src/script/cpp_api/s_security.cpp b/src/script/cpp_api/s_security.cpp
index 0d989ae04..4e2ec8062 100644
--- a/src/script/cpp_api/s_security.cpp
+++ b/src/script/cpp_api/s_security.cpp
@@ -638,6 +638,10 @@ bool ScriptApiSecurity::checkWhitelisted(lua_State *L, const std::string &settin
 		return false;
 	std::string mod_name = readParam<std::string>(L, -1);
 
+	// Builtin can access anything
+	if (mod_name == BUILTIN_MOD_NAME)
+		return true;
+
 	std::string value = g_settings->get(setting);
 	value.erase(std::remove(value.begin(), value.end(), ' '), value.end());
 	auto mod_list = str_split(value, ',');

```

For now I suppose you could use the insecure environment to copy the `modpack.conf` to a file in a readable location, then read that file with `Settings()`.

Reply:
Why are we discussing an approach that works around the engine when we could just add an API that returns the exact game/modpack structure in use?